nautobot/ipam/utils/migrations.py

```diff
@@ -731,3 +731,27 @@ def ensure_correct_prefix_broadcast(apps):
                 )
             prefix.broadcast = true_broadcast
             prefix.save()
+
+
+def increment_names_of_records_with_similar_names(model: models.Model):
+    """
+    This function increments the names of records with similar names in a given model.
+    """
+    cache = set()
+    records_to_update = []
+    for instance in model.objects.all():
+        name = instance.name
+        counter = 1
+        while name in cache:
+            max_name_length = model.name.field.max_length
+            name = f"{instance.name[:max_name_length]} {counter}"
+            counter += 1
+
+        if name != instance.name:
+            print(f'{model._meta.verbose_name} instance {instance.id} is being renamed to "{name}" for uniqueness')
```

Comment:
Since this will typically be seen in the context of a migration, might be good to prepend `    ` to the printed string so that it appears nested under the migration message.